internal/gitinterface/sync.go

```diff
@@ -64,7 +64,7 @@ func (r *Repository) Fetch(remoteName string, refs []string, fastForwardOnly boo
 }
 
 func (r *Repository) FetchObject(remoteName string, objectID Hash) error {
-	args := []string{"fetch", "--depth=1", remoteName, objectID.String()}
+	args := []string{"fetch", remoteName, objectID.String()}
```

Comment:
`depth=1` results in `objectID` being fetched but as a "graft", meaning we can't reason about whether it's ahead or behind other objects.

Reply:
this was a real head scratcher when writing tests because merge-base --is-ancestor reported "false" when asked if the parent commit was the ancestor of the test commit.